infrastructure/pyproject.toml

```diff
@@ -0,0 +1,8 @@
+[project]
+name = "infrastructure"
+version = "0.1.0"
+requires-python = ">=3.12"
```

Comment:
Is it okay that this doesn't match the root `requires-python` version?

Reply:
yea, this should be fine, the workspace should resolve properly since `>=3.12` is a subset of `>=3.10` and wouldn't conflict, I have it at 3.10 in part just because most ML stuff gets weird with newer versions, but, if we are using flyte then it actually doesn't matter, because those will be containerized functions, so if we need `3.10` for `pandas` in one spot, `3.9` for some `numpy` reason, we do it in the image definitions, we are not actually evaluating it

so basically, the answer to your question is it's fine, but also why not just have both be newer versions of python if we can, I updated the root `pyproject.toml`